opencti-platform/opencti-graphql/src/modules/internal/document/document-domain.ts

```diff
@@ -119,12 +121,36 @@ export const allRemainingFilesCount = async (context: AuthContext, user: AuthUse
   return elCount(context, user, [READ_INDEX_INTERNAL_OBJECTS], remainingOpts);
 };
 
+export const checkFileAccess = async (context: AuthContext, user: AuthUser, scope: string, { entity_id, filename, id }: { entity_id?: string, filename: string, id: string }) => {
+  if (isEmptyField(entity_id)) {
+    return true;
+  }
+  const userInstancePromise = internalLoadById(context, user, entity_id);
+  const systemInstancePromise = internalLoadById(context, SYSTEM_USER, entity_id);
+  const [instance, systemInstance] = await Promise.all([userInstancePromise, systemInstancePromise]);
+  if (isEmptyField(instance)) {
+    if (isNotEmptyField(systemInstance)) {
+      const data = buildContextDataForFile(systemInstance as BasicStoreObject, id, filename);
+      await publishUserAction({
+        user,
+        event_type: 'file',
+        event_scope: scope,
+        event_access: 'extended',
+        status: 'error',
+        context_data: data
+      } as UserAction);
+    }
+    throw ForbiddenAccess('Access to this file is restricted', { id: entity_id, file: id });
+  }
+  return true;
+};
+
 // Get Files paginated with auto enrichment
 // Images metadata for users
 // In progress virtual files from export
-export const paginatedForPathsWithEnrichment = async (context: AuthContext, user: AuthUser, paths: string[], opts?: FilesOptions<BasicStoreEntityDocument>) => {
+export const paginatedForPathsWithEnrichment = async (context: AuthContext, user: AuthUser, path: string, entity_id: string, opts?: FilesOptions<BasicStoreEntityDocument>) => {
```

Comment:
if it's now singular "path" instead of "paths", maybe we can rename the method too ? **paginatedForPathWithEnrichment** or **paginatedFilesWithEnrichment** so it's not dependent on the parameters